sdk/core/System.ClientModel/src/Convenience/ClientResultOfT.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.ClientModel.Primitives;
+using System.ComponentModel;
+
+namespace System.ClientModel;
+
+public class ClientResult<T> : OptionalClientResult<T>
+{
+    protected internal ClientResult(T value, PipelineResponse response)
+        : base(value, response)
+    {
+        // Null values must use OptionalClientResult<T>
+        ClientUtilities.AssertNotNull(value, nameof(value));
```

Comment:
I prefer the semantics when this was represented as `Argument.AssertNotNull`.   Having it in a general utilities class leaves it more ambiguous and it's not clear that we intend it to be used for validating arguments.

Reply:
I'm happy to noodle this a bit - a couple of suggestions --

First, context: I coped operations from several different internal shared source types into ClientUtilities when this was a public type for the demo so we wouldn't have multiple utility types floating around.  After the demo, I made this an internal type since we wanted to postpone designing and getting BCL-team feedback on it as a public API, and we agreed the generator would generate single-line argument null-checks to move away from internal shared source for ClientModel.  But @KrzysztofCwalina is in favor of trying to make this a public type; we would just have to do the work to decide what goes in/what's out; what the type name is, etc.  I have an issue tracking this here: https://github.com/Azure/azure-sdk-for-net/issues/41035, which we can break into multiple issues if that's preferred.

With that - for now, we have some options -- 
1. We could change the argument validation method names to AssertArgumentXx
2. We could break ClientUtilities into multiple classes again to reflect the functional separation they have today in Azure.Core.  As long as they stay internal, we can give them the same names as the Azure.Core types without conflict, I think -- we couldn't do that when we needed them to be public for the demo.

Do you have a preference for which way to go for the purposes of this PR, @jsquire?

Reply:
Per later comment I will break these into types with names that better match the Azure.Core types.